contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,190 @@
+
+* **Native Apps** - Angular can be used to build <tooltip content="a smartphone application that is coded in a specific programming language, such as Swift for iOS or Java for Android operating systems">native mobile applications</tooltip>. There are many frameworks such as [NativeScript](https://www.nativescript.org/) and [Ionic](https://ionicframework.com/)
+that can be integrated with Angular for mobile application development. The [official Ionic website](https://ionicframework.com/docs/intro) claims that "integration with the Angular ecosystem is a breeze".
+
+* **Desktop Apps** - With Angular you can create desktop applications across Mac, Windows and Linux systems. A common approach used by many developers is to use [Electron](https://electronjs.org/), a framework maintained by Github. You can install the Electron framework in your Angular project with the help of the CLI and start building cross-platform applications.
+
+### Benefit 3: Angular provides two-way data binding.
+
+As explained previously, Angular divides the application into the Model, View and Controller architecture. Data binding establishes the connection between the Model and View components. There are two types of data binding as explained below.
+
+![Data Binding](one_two.png)
+
+Angular uses two-way data binding as opposed to popular tools like React that use one way data binding. How does this help?
+
+It ensures the exchange of data between the View and Model in a bi-directional manner. If the User Input changes, data in the backend is updated by Angular. Additionally, if the Model state (data) changes, the UI is updated automatically. This means reduced coding for developers.
+
+One-way data binding (used in React) involves uni-directional communication. If the model is changed, the changes are reflected in the UI. However, if the UI element changes the model state is not updated. Developers must provide this synchronization code. 
+
+This [article](https://medium.com/@preethi.s/angular-custom-two-way-data-binding-3e618309d6c7) on Medium provides a good introduction to two-way data binding in Angular.
+
+### When not to use Angular?
+
+Like any other framework/library, Angular has its share of disadvantages.
+
+1. **Steep Learning curve** - Angular requires you to learn many concepts, such as directives, modules, components, services, dependency injection, pipes, and most importantly TypeScript. 
+The large number of new concepts can be confusing to newcomers.
+
+2. **Opinionated Framework** - Angular is opinionated about how you structure your code. This means it provides you with a set of tools to build your application. Angular expects you to use these tools as much as you can. While libraries like React allow you to integrate any third party services in your application.
```

Comment:
`This means it provides you with a set of tools to build your application.` hmmm... that's not an explanation of _opinionated_. It needs to connect with the subsequent sentence to give the full meaning.

Reply:
This sounds like Angular is_opinionated_ about what tools you use only. Not other things such as conventions, structure, ... Is that the case?